table/plain/plain_table_reader.cc

```diff
@@ -183,6 +183,8 @@ Status PlainTableReader::Open(
     // can be used.
     new_reader->full_scan_mode_ = true;
   }
+  // PopulateIndex can add to the props, so don't store them until now
+  new_reader->table_properties_.reset(props);
```

Comment:
Should we move the same line inside PopulateIndex()? It feels confusing to me to set it twice but happens to be the same one to be correct.

Reply:
Old state: only set in PlainTableReader::PopulateIndex, which is not always called. New state: only/always set in PlainTableReader::Open (and *Test*PlainTableReader ctor since it does not use PlainTableReader::Open).